posts/2023-10-30_floating_point/floating_point.qmd

```diff
@@ -0,0 +1,338 @@
+---
+title: "Floating point"
+author:
+  - name: Stefan Thoma
+description: "The untold story of how admiral deals with floating points."
+date: "2023-10-30"
+# please do not use any non-default categories.
+# You can find the default categories in the repository README.md
+categories: [admiral]
+# feel free to change the image
+image: "admiral.png"
+---
+
+<!--------------- typical setup ----------------->
+
+```{r setup, include=FALSE}
+long_slug <- "2023-10-30_floating_point"
+# renv::use(lockfile = "renv.lock")
+```
+
+<!--------------- post begins here ----------------->
+```{r, echo=FALSE}
+# check if version 444 of admiral is installed, and throw error if not:
+if (packageVersion("admiral") < "0.12.2") {
+  stop("Please install a more recent version of admiral.
+       At least version 0.12.2 is required.")
+}
+```
+
+```{r, echo=FALSE}
+```
+
+[{{admiral}}](https://github.com/pharmaverse/admiral) recently ran into some trouble when dealing with floating point values, captured [by this thread on GitHub](https://github.com/pharmaverse/admiral/pull/2060).
+This post gives a brief overview on floating point values, recaps the discussion on GitHub, and explains how [{{admiral}}](https://github.com/pharmaverse/admiral) deals with floating point values.
+
+## Floating point values
+
+Floating point values are numeric objects representing numbers between integers, e.g. 0.5., 2.3, 3.1415, etc.
+However, floating point numbers are not stored like integers, and most floating point numbers are approximations to the number they represent.
+To see what value a floating point number is actually stored as, we can use the `format()` function where we can increase the number of digits shown:
+
+```{r, echo=FALSE, message = FALSE}
+library(dplyr)
+format(1.4, digits = 22)
+```
+
+These very small numerical differences impact the result of mathematical operations:
+
+```{r}
+0.1 + 0.2 == 0.3
+```
+
+If we look at the actually stored values, this makes sense:
+
+```{r}
+0.1 %>% format(digits = 22)
+0.2 %>% format(digits = 22)
+
+(0.1 + 0.2) %>% format(digits = 22)
+
+0.3 %>% format(digits = 22)
+```
```

Comment:
I like to suppress the `[1]` for small outputs.  I think you can set comment = "" in the code chunk and it will remove them.  
![image](https://github.com/pharmaverse/blog/assets/10111024/acc0533d-d7a3-48b9-806f-f39fe264e0b3)

Reply:
Eh this does not work! 
I'll leave as is for now.